src/physics/utils/pio_reader.F90

```diff
@@ -0,0 +1,751 @@
+        character(len=*),    intent(out) :: errmsg
+        
+        character(len=cl)  :: local_file_path  !NetCDF file path on local file system
+
+        if(this%sima_pio_fh%is_file_open) then
+          errcode = 1
+          errmsg = "Trying to reuse pio_reader already used for: '"//this%sima_pio_fh%file_path
+        end if
+
+        if(file_path .eq. 'UNSET_PATH') then
+         errcode = 1
+         errmsg = "Found UNSET_PATH trying to open file"
+        end if
+  
+        call cam_get_file(file_path, local_file_path)
+        call cam_pio_openfile(this%sima_pio_fh%pio_fh, local_file_path, PIO_NOWRITE)
+  
+        !Set file handle metadata
+        this%sima_pio_fh%file_path    = local_file_path
+        this%sima_pio_fh%is_file_open = .true.
+  
+        !File was successfully opened
+        errcode = 0
+        errmsg = ''
+    end subroutine open_netcdf_file
+
+    subroutine close_netcdf_file(this, errcode, errmsg)
+        use pio, only: pio_closefile
+
+        class(pio_reader_t), intent(inout)  :: this
+        integer,             intent(out) :: errcode
+        character(len=*),    intent(out) :: errmsg
+
+        !Close NetCDF File:
+        call pio_closefile(this%sima_pio_fh%pio_fh)
+
+        !Inidcate that file handle array id is no longer in use:
+        this%sima_pio_fh%is_file_open = .false.
+        this%sima_pio_fh%file_path = ''
+
+        !File was successfully closed
+        errcode = 0
+        errmsg = ''
+    end subroutine close_netcdf_file
+
+    subroutine get_netcdf_var_int(this, varname, var, errcode, errmsg)
+        use ccpp_kinds, only: kind_phys
+        use pio,        only: pio_inq_varid
+        use pio,        only: pio_inq_dimlen
+        use pio,        only: pio_inquire_variable
+        use pio,        only: pio_seterrorhandling
+        use pio,        only: pio_get_var
+        use pio,        only: PIO_NOERR
+        use pio,        only: PIO_BCAST_ERROR
+        
+        class(pio_reader_t), intent(in)  :: this
+        character(len=*),    intent(in)  :: varname
+        integer, pointer,    intent(out) :: var(..) !Character variable that file data will be read to.
+        integer,             intent(out) :: errcode !Error code
+        character(len=*),    intent(out) :: errmsg  !Error message
+
+        !Local variables:
+        type(file_desc_t)    :: pio_file_handle !File handle type used by PIO
+        character(len=cl)    :: file_path       !Path to NetCDF file
+        integer              :: err_handling    !PIO error handling code
+        integer              :: var_id          !NetCDF variable ID
+        integer              :: ndims           !Number of variable dimensions on NetCDF file
+        integer, allocatable :: dim_ids(:)      !Variable dimension IDs
+        integer, allocatable :: dim_sizes(:)    !Variable dimension sizes
+  
+        integer :: i !loop control variable
+        !----------------------
+  
+        !Extract open file information:
```

Comment:
Should there be a check that `is_file_open == .true.`? (similar comment for remaining functions)

Reply:
Yes, good catch!  Added checks to all read subroutines.